syntax = "proto3";

package peek;

service peekService {
    rpc addUser (userRequest) returns (userResponse);
    rpc loginUser (userRequest) returns (userResponse);
    rpc getCachedData (peekRequest) returns (peekResponse);
    rpc getNewData (peekRequest) returns (peekResponse);
}

message userRequest {
    string username = 1;
    string password = 2;
}

message userResponse {
    int32 userID = 1;
}

/* appId
0 = Gmail
1 = Instagram
*/
message peekRequest {
    enum app {
        GMAIL = 0;
        INSTAGRAM = 1;
        DISCORD = 2;
        WHATSAPP = 3;
        MESSENGER = 4;
        TWITTER = 5;
    }

    int32 userID = 1;
    app appID = 2;
}

message peekResponse {
    repeated topic topics= 1;
}

message topic {
    string name = 1;
    string emoji = 2;
    string highlight = 3;
    string summary = 4;
    repeated notif notifs = 5;
}

message notif {
    string title = 1;
    string uri = 2;
}